### DISCUSSION TITLE
watchdog & breakpoints

### DISCUSSION THREAD

**Discussion by tve** [Moddable-OpenSource/moddable#1255] [Q&A]
I like to have some form of watchdog functionality in my app code that ensures that certain tasks keep running. The esp-idf has some convenient watchdog functionality where one can create a "user" watchdog and then "feed" it in one's code. Only problem is, if I set a breakpoint my code no longer feeds the watchdog and a reset results. Is there a good place to hook into the breakpoint functionality so the watchdogs can be disabled when stopping and re-enabled when running again? Or a different suggestion?

**Comment by phoddie** [comment]
Yea, this is one reason that the watchdog is disabled in debug builds. In theory, it can work though. ESP8266 always has the watchdog enabled, so the host has to deal with it. [Here's where it does it](https://github.com/Moddable-OpenSource/moddable/blob/1918979ced6d36f3a4c380ade146c083beaeb084/build/devices/esp/main.cpp#L141-L162), by calling `system_soft_wdt_feed` in low level functions for the serial I/O for the debug connection. You should be able to do something similar on ESP32. It could be conditionalized on whether the watchdog is enabled, so it wouldn't have a runtime cost by default.
